security/fma-generic-contracts.md

```diff
@@ -0,0 +1,147 @@
+
+- [Generic items we need to take into account for any smart contract upgrade:](#generic-items-we-need-to-take-into-account-for-any-smart-contract-upgrade)
+  - [Proxy Initialization Failure](#proxy-initialization-failure)
+  - [New Implementation Overwrites (wrong) Slots](#new-implementation-overwrites-wrong-slots)
+  - [Proxy Initialized to Wrong Values](#proxy-initialized-to-wrong-values)
+  - [Can’t Upgrade Implementation](#cant-upgrade-implementation)
+  - [Able to Reinitialize Implementation](#able-to-reinitialize-implementation)
+  - [Backwards-incompatible ABIs](#backwards-incompatible-abis)
+  - [Invalid `DisputeGameFactory.setImplementation` execution](#invalid-disputegamefactorysetimplementation-execution)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+### Generic items we need to take into account for any smart contract upgrade:
+
+#### Proxy Initialization Failure
+
+- **Description:**
+  Unable to initialize the proxy due to a logic error.
+
+- **Risk Assessment:**
+  High severity since it allows arbitrary initialization.
+  Low likelihood since the initialization is a minimal operation.
+
+- **Mitigations:**
+  Extensive unit and end-to-end testing for proxy initializations.
+  Proxy initializations are verified through simulations prior to execution.
+
+- **Detection:**
+  Call initialize(..args) or by simulating the effort.
+
+- **Recovery Path(s):**
+  Perform an upgrade to perform the initialization.
+
+
+#### New Implementation Overwrites (wrong) Slots
+
+- **Description:**
+  An upgrade to a smart contract that inadvertently overwrites the wrong storage values in proxy.
+
+- **Risk Assessment:**
+  High severity due to arbitrary slot bounds.
+  Low likelihood since this is well looked out for.
+
+- **Mitigations:**
+  Storage layouts are validated to only extend and not rewrite the slots of the previous version.
+
+- **Detection:**
+  If this error were missed, it could be detected by looking at the storage values in the proxy or recognizing an inconsistency in variables stored in overwritten slots.
+  Alternatively, we could detect it from monitoring or user reports that something seems off/incorrect.
+
+- **Recovery Path(s):**
+  Deploy new implementation with a fixed storage layout and write new values to storage.
+  Requires an upgrade.
+
+
+#### Proxy Initialized to Wrong Values
+
+- **Description:**
+    The proxy is initialized with the wrong values.
+
+- **Risk Assessment:**
+  High severity since arbitrary values can be incorrectly used.
+  Low likelihood since proxy initialization has a small surface area and is well verified/tested.
+
+- **Mitigations:**
+  Validate initialized values through extensive unit testing.
+  Simulations prior to performing the upgrade, verifying the initialized values.
+
+- **Detection:**
+  A simulation can show an unintended initialized proxy value.
+  Monitoring or user reports.
```

Comment:
Agreed, but keeping this PR to a port of the notion doc to unblock the stage 1.4 FMA. A follow on PR should address this. I'd recommend putting up a ticket in this repository to track this suggestion and implementing the changes once this pr lands.

Reply:
Unresolving for the same reason as https://github.com/ethereum-optimism/design-docs/pull/140#discussion_r1821103827, since this is a valid detection for the stage 1.4 FMA ad that FMA depends on this PR :)

Reply:
I'm very unfamiliar with the workings of "user reports" so I'm not able to answer these questions.

Reply:
There's no concrete thing here, @Ethnical is just saying that some user reports come through immunefi issues, others get raised through various other means. Something like this seems fine
```suggestion
  Monitoring or user reports, including Immunefi reports and other means of getting user feedback
```